### ISSUE TITLE
YOLOv8 as backend for Label Studio, including active learning and online learning

### ISSUE THREAD

**Issue by ruzko** [ultralytics/ultralytics#3927]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

For [my project](https://gitlab.com/papiris/predator-detect-and-notify) I use YOLOv8.

I've got quite a few datasets I want to label in order to train the model, and I currently use [YOLO as a pre-labeling backend](https://gitlab.com/papiris/label-studio-backend-yolov8-animals) for Label Studio to get this done. This means manually exporting the labeled dataset and running the train method with the path to the yaml defined.
What I would like to do, is use YOLO for [online learning and active learning in Label Studio](labelstud.io/guide/ml.html) instead.
I found [this repo ](https://github.com/4rn3/yolov7_labelstudio/blob/main/model_backend.py) which uses YOLO v7 in an approximately online learning method, but I've been unable to adapt it to fit v8. It also seems a bit wasteful to move files around so much.

Can a v8 model be used like this:
1. Model is loaded using existing semi-trained weights
2. Model pre-labels some images
3. Human corrects/confirms labeling/annotations in Label Studio
4. Model has its training 'resumed' for a few more epochs using the newly human-corrected annotations

Repeat steps 1-4 until desired mAP is reached✨

### Additional

❤️ yolo

**Comment by glenn-jocher** [comment]
Dear @ruzko,

Thanks for reaching out and sharing your project. YOLOv8 is indeed capable of the workflow you're envisioning. Let's break it down:

1. **Load the model using existing semi-trained weights**: Absolutely, YOLOv8 has been designed to start training from pre-existing weights. Once you have a model partially trained, you can use that checkpoint to resume its training.

2. **Pre-label some images**: Yes, you can use this semi-trained model to make predictions on your images, essentially acting as an initial round of pre-labeling.

3. **Human corrects/confirms labeling/annotations in Label Studio**: This is more dependent on your Label Studio workflow, but generally, you can export these pre-labeled images to Label Studio where you can then correct and confirm these labels.

4. **Resume training for a few more epochs using the human-corrected annotations**: YOLOv8 supports continued training from any given checkpoint. Once you have your new, human-corrected training set, you can resume training your model for more epochs using this data.

As you mentioned, the key will be in integrating this workflow into your current software architecture. The steps you've laid out seem perfectly feasible, and I encourage your experimentations with adapting them into your process.

Just remember to keep track of your number of epochs, learning rate, and other relevant training hyperparameters as you iterate in this semi-supervised learning manner.

Best of luck with your work, it sounds like an exciting application for YOLOv8! 💫

**Comment by paulinavelasquez** [comment]
Yes, I just created a function to do it manually. Only since it was displayed in the terminal I thought maybe I could get them in variables. Thank you very much for the quick responses. Greetings.
